fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -100,53 +100,63 @@ public void createCatalog(CreateCatalogStmt stmt) throws DdlException {
+                    } else {
+                        Authorizer.getInstance().setAccessControl(catalogName, new NativeAccessController());
+                    }
                 } else {
-                    Authorizer.getInstance().setAccessControl(catalogName, new NativeAccessController());
+                    Authorizer.getInstance().setAccessControl(catalogName, new RangerHiveAccessController(serviceName));
                 }
-            } else {
-                Authorizer.getInstance().setAccessControl(catalogName, new RangerHiveAccessController(serviceName));
-            }
 
-            // TODO：please keep connector and catalog create together, they need keep in consistent asap.
-            CatalogConnector connector = connectorMgr.createConnector(new ConnectorContext(catalogName, type, properties), false);
-            if (null == connector) {
-                LOG.error("{} connector [{}] create failed", type, catalogName);
-                throw new DdlException("connector create failed");
+                CatalogConnector connector = connectorMgr.createConnector(
+                        new ConnectorContext(catalogName, type, properties), false);
+                if (null == connector) {
+                    LOG.error("{} connector [{}] create failed", type, catalogName);
+                    throw new DdlException("connector create failed");
+                }
+                long id = isResourceMappingCatalog(catalogName) ?
+                        ConnectorTableId.CONNECTOR_ID_GENERATOR.getNextId().asInt() :
+                        GlobalStateMgr.getCurrentState().getNextId();
+                Catalog catalog = new ExternalCatalog(id, catalogName, comment, properties);
+                catalogs.put(catalogName, catalog);
+
+                if (!isResourceMappingCatalog(catalogName)) {
+                    GlobalStateMgr.getCurrentState().getEditLog().logCreateCatalog(catalog);
+                }
+            } catch (StarRocksConnectorException e) {
+                LOG.error("connector create failed. catalog [{}] ", catalogName, e);
+                throw new DdlException(String.format("connector create failed {%s}", e.getMessage()));
+            } finally {
+                writeUnLock();
             }
-            long id = isResourceMappingCatalog(catalogName) ?
-                    ConnectorTableId.CONNECTOR_ID_GENERATOR.getNextId().asInt() :
-                    GlobalStateMgr.getCurrentState().getNextId();
-            Catalog catalog = new ExternalCatalog(id, catalogName, comment, properties);
-            catalogs.put(catalogName, catalog);
-
-            if (!isResourceMappingCatalog(catalogName)) {
-                GlobalStateMgr.getCurrentState().getEditLog().logCreateCatalog(catalog);
+        } catch (Exception e) {
+            if (connectorMgr.connectorExists(catalogName)) {
+                connectorMgr.removeConnector(catalogName);
```

Comment:
should add lock here

Reply:
`ConnectorExists` have locks

Reply:
How do you make sure remove connector and catalog atomically?